### PR TITLE
Include information about App Launch feature to SAD

### PR DESCRIPTION
**Pull request by LevchenkoS**
Has been done:
- Added information about App Launch component;
- App Launch added to:
  - Component view diagram;
  - Modules and Code Base Organization diagram;
  - Component-to-Layer table;
- Added new diagram to Component
  Interaction View.

Related to: [APPLINK-24897](https://adc.luxoft.com/jira/browse/APPLINK-24897)

@pestOO, please review.

### CODE REVIEW

**Comment by justinjdickow** [comment]
Preview this pull request at https://smartdevicelink.com/en/guides/pull_request/0b755f7f6e39f8eaa7e02ac2e3d7cd4492b47a18/overview/

File: docs/Software Architecture Document/Component Interaction View/assets/app_launch.plantuml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+== Device connected ==
+
+CH -> AM : Notification
+AM -> AL : Device Connected
+AL -> AL : Get apps data for device
+loop 
+	AL -> Resum : Get HMI level for app
+	AL <-- Resum 
+end
+loop
```
Comment (pestOO) [inline docs/Software Architecture Document/Component Interaction View/assets/app_launch.plantuml]: @LevchenkoS,
Maybe better to merge it in one loop?
And add comment like 'for each saved app'.

**Comment by LevchenkoS** [comment]
@LuxoftAKutsan, @AGritsevich, please review.

**Comment by AGritsevich** [comment]
reviewed

**Comment by justinjdickow** [comment]
Preview this pull request at https://smartdevicelink.com/en/guides/pull_request/0abba45f5d192cbc4e98c52a9de0b1bf3cdf4313/overview/

File: docs/Software Architecture Document/Component Interaction View/assets/app_launch.plantuml (hunk 1 above)
Comment (LevchenkoS) [inline docs/Software Architecture Document/Component Interaction View/assets/app_launch.plantuml]: @pestOO, fixed in 0abba45.

**Comment by pestOO** [comment]
@AnastasiyaBritanova, please review

**Comment by LuxoftAKutsan** [comment]
Reviewed

File: docs/Software Architecture Document/Components View/index.md [hunk 2]
```diff
@@ -136,6 +136,19 @@ The view is represented by module and subsystem diagrams that show the system's
...
+    - Launch known applications on devices.
+  - *Relations*
+    - Composed by ***Application Manager*** 
+    - Use ***Resume Controller*** interface to get HMI level of saved application.
+  - *Interfaces*
+    - Provides ***App Launch Controller*** interface 
+  - *Behavior*
+    - ***App Launch*** launch all known applications on newly connected device.
+  - *Constraints*
+    - N/A
```
Comment (AnastasiyaBritanova) [inline docs/Software Architecture Document/Components View/index.md]: there are constrains, actually: appLaunch should not work 1) for Android apps + 2) for apps connected via version lower than 4.

Comment (LevchenkoS) [inline docs/Software Architecture Document/Components View/index.md]: @AnastasiyaBritanova, sorry, but what exactly should have version higher then 4?

Comment (AnastasiyaBritanova) [inline docs/Software Architecture Document/Components View/index.md]: @LevchenkoS sorry :) sdl protocol version was meant.

File: docs/Software Architecture Document/Component Interaction View/assets/app_launch.plantuml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+@startuml
```
Comment (AnastasiyaBritanova) [inline docs/Software Architecture Document/Component Interaction View/assets/app_launch.plantuml]: @LevchenkoS is it further used? did not find you reference this diagram at the page.

Comment (LevchenkoS) [inline docs/Software Architecture Document/Component Interaction View/assets/app_launch.plantuml]: @AnastasiyaBritanova, this is the source code for `app_launch.png` image which included to `Component Interaction View`.

**Comment by justinjdickow** [comment]
Preview this pull request at https://smartdevicelink.com/en/guides/pull_request/ec7314dd61bfbdcc9d2d91f244df289c5ed07586/overview/

File: docs/Software Architecture Document/Components View/index.md (hunk 2 above)
Comment (LevchenkoS) [inline docs/Software Architecture Document/Components View/index.md]: @AnastasiyaBritanova, fixed in ec7314d.

**Comment by pestOO** [comment]
@LevchenkoS,
Please, wait for #9 merge, then rebase and update history chapter,

**Comment by LevchenkoS** [comment]
@AnastasiyaBritanova, please, finish review.

**Comment by AnastasiyaBritanova** [comment]
@LevchenkoS reviewed

**Pull request merged**